java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -478,14 +478,15 @@ public void reAlloc() {
    * @throws OutOfMemoryException if the internal memory allocation fails
    */
   public void reallocDataBuffer() {
-    long baseSize = lastValueAllocationSizeInBytes;
     final int currentBufferCapacity = valueBuffer.capacity();
-
-    if (baseSize < (long) currentBufferCapacity) {
-      baseSize = (long) currentBufferCapacity;
+    long newAllocationSize = currentBufferCapacity * 2;
+    if (newAllocationSize == 0) {
```

Comment:
is it worth trying to consolidate this logic?

Reply:
I tried doing this but it's becoming messy since the usage in the three places is slightly different -
1. In BaseFixedWidthVector - the computation is based on element count, so it's safe to use int
2. In BaseVariableWidthVector#reallocDataBuffer - the computation is based on buffer size (instead of element count), so we need to use longs
 3. In BaseVariableWidthVector#reallocValidityAndOffsetBuffers - the computation is based on element count, but there is the extra +1

If I switch all of them to use long, the change needs to percolate to a lot other external/internal APIs. I'd prefer to keep it this way.

Reply:
Thanks for trying.